components/org.wso2.micro.integrator.initializer/src/main/java/org/wso2/micro/integrator/initializer/handler/transaction/TransactionCountHandlerComponent.java

```diff
@@ -0,0 +1,184 @@
+
+    private static final String TRANSACTION_CONFIG_SECTION = "transaction_tracker";
+    private static final String TRANSACTION_CONFIG_DATA_SOURCE = "datasource";
+    private static final String TRANSACTION_CONFIG_SCHEDULED_PERIOD = "update_interval";
+
+    private ScheduledExecutorService txCountWriterTaskScheduler;
+
+    public void start(DataSourceService dataSourceService) throws TransactionCounterException, DataSourceException {
+        DataSource dataSource = getTransactionDataSource(dataSourceService);
+
+        // initialize cipher for encryption needs.
+        Cipher cipher = CryptoUtil.initializeCipher();
+
+        String nodeId = generateRandomId();
+
+        // initialize Transaction store and adding to the data-holder.
+        TransactionStore transactionStore = new TransactionStore(dataSource, nodeId, cipher);
+        DataHolder.getInstance().setTransactionStore(transactionStore);
+
+        // start and schedule transaction writer task.
+        scheduleTransactionWriterTask(transactionStore);
+    }
+
+    /**
+     * Get the transaction count update period for running the transaction writer scheduled task.
+     *
+     * @return - transaction count update period.
+     */
+    private int getTransactionCountScheduledPeriod() {
+        Object scheduledPeriodObject = ConfigParser.getParsedConfigs().get(
+                TRANSACTION_CONFIG_SECTION + "." + TRANSACTION_CONFIG_SCHEDULED_PERIOD);
+        if (null != scheduledPeriodObject) {
+            String scheduledPeriodStr = scheduledPeriodObject.toString();
+            try {
+                return Integer.parseInt(scheduledPeriodStr);
+            } catch (NumberFormatException e) {
+                LOG.warn("Cannot parse the provided transaction count update period " + scheduledPeriodStr
+                                 + " as an Integer. Hence, proceeding with default value ("
+                                 + DEFAULT_SCHEDULED_PERIOD_INTERVAL + "s)");
+                return DEFAULT_SCHEDULED_PERIOD_INTERVAL;
+            }
+        } else {
+            LOG.debug("Transaction count update period is set to default value (" + DEFAULT_SCHEDULED_PERIOD_INTERVAL
+                              + "s).");
+            return DEFAULT_SCHEDULED_PERIOD_INTERVAL;
+        }
+    }
+
+    /**
+     * Schedule the transaction writer task to write the current transaction count to the database at a fixed rate.
+     *
+     * @param transactionStore - transactionStore instance.
+     */
+    private void scheduleTransactionWriterTask(TransactionStore transactionStore) {
+        txCountWriterTaskScheduler = Executors.newSingleThreadScheduledExecutor();
+        int scheduledPeriod = getTransactionCountScheduledPeriod();
+
+        txCountWriterTaskScheduler.scheduleAtFixedRate(() -> {
+            try {
+                transactionStore.addTransaction();
+            } catch (TransactionCounterException e) {
```

Comment:
I think we should catch throw able here, if some other exception happens, the schedule task will be stopped thereafter. @sdkottegoda

Reply:
+1